### ISSUE TITLE
[Feature Request] D1X: Option to Replace (or remove) Vulcan Drillers and Red Hulks

### ISSUE THREAD

**Issue by narzepatel** [dxx-rebirth/dxx-rebirth#645]
In a similar manner to the option to remove or nerf Thief Bots in D2X, I want to request for **an option** to do something similar with the Vulcan Driller and Red Hulks in D1X. Either an option to make them weaker or replacing them outright with another enemy type.

**Comment by vLKp** [comment]
Eliminating the thief bot is comparatively easier.  As far as I know, it never had anything of value prior to stealing the player's equipment, so simply deleting it from the level had no negative effects.  There was only one of it, and the internal special logic required to make it work also made it easy to identify for removal.  Vulcan Drillers and Red Hulks sadly have none of these virtues.  They are a somewhat regular enemy type, which may well carry valuable gear you want to capture.  They have no special recognition in the code, so any patch to affect them would need to hardcode that "Robot 10" (or whichever ID they are) gets specially handled.

None of this is to say I won't do what you're asking, but it's not as simple as it was to remove the thief, particularly since you want them weakened rather than deleted.

**Comment by narzepatel** [comment]
Well how about having the drillers fire small fast projectiles that can be dodged like D2, or something like reducing the health of red hulks or making their homing missiles stop following once the lock is avoided.

**Comment by vLKp** [comment]
- Changing the projectiles is approximately as hard as changing the health of the robot.
- Changing the health is supported today, but only through an interface that is not particularly user friendly.  You would need to define a robot mod.  You might find `d2x-rebirth/utilities/hamjson.py` useful, as it would make defining the alternate robot slightly easier.  To change total health, modify the `strength` field.
- Homing missiles already stop following once their lock is lost.  However, if they subsequently see you again, they can reacquire lock and resume following.

**Comment by guillaumechereau** [comment]
I agree with @narzepatel.  Specially the drillers are really not fun in my opinion, and almost ruin an otherwise excellent game.  I am not sure how feasible this is, but maybe an option to add a small delay before they start shooting would be nice.

**Comment by vLKp** [comment]
Making _any_ change to their behavioural logic first requires that the game can distinguish the "bad" robots (Driller, Red Hulk) from other robots (Green Hulk, Brown Hulk, etc.).  Once the ability to identify robots is present, all the requested data changes become relatively easy.  Until that is present, the game cannot determine which robots should be weakened and which should be kept at full strength.  Currently, there is no generic mechanism to identify bad robots versus other robots.
